### DISCUSSION TITLE
[Beginner]Not able to switch to Twitter authentication popup window.

### DISCUSSION THREAD

**Discussion by ChowKwanhwa** [seleniumbase/SeleniumBase#2509] [General]
I am trying to create a automation script that involves Twitter authentication. Ideally, the script should be able to detect the authentication popup window, and send username and password to the input fields. 
However, it looks like the script couldn't detect the popup window. 
```
from seleniumbase import SB
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

def open_the_turnstile_page(sb):
    sb.driver.uc_open_with_reconnect(
        "https://whitelist.personajourney.io?r=v3tx8t", reconnect_time=2.5,)
    element1 = sb.wait_for_element_visible("div.css-1qyct49")
    sb.driver.execute_script("arguments[0].scrollIntoView(true);", element1)
    sb.assert_element("div.css-1qyct49")
    sb.click("div.css-1qyct49")
    
    # Wait for element2 to become visible with a timeout of 20 seconds
    element2 = sb.wait_for_element_visible("div.css-llps6v", timeout=20)
    sb.driver.execute_script("arguments[0].scrollIntoView(true);", element2)
    sb.assert_element("div.css-llps6v")  
    sb.click("div.css-llps6v")            
    
    # Check if there are multiple window handles available
    if len(sb.driver.window_handles) > 1:
        # Switch focus to the popup window
        popup_window = sb.driver.window_handles[-1]
        sb.driver.switch_to.window(popup_window)

        # Print the URL of the popup window
        popup_url = sb.driver.current_url
        print("Popup window URL:", popup_url)
    else:
        print("No popup window detected.")

with SB(uc=True, test=True, headless=False) as sb:
    open_the_turnstile_page(sb)
```
        

 It shows No popup window detected. on the console. 
 
 Thanks:)

**Comment by mdmintz** [comment]
Not sure what you're doing because there's no Twitter authentication section on the URL you gave. Also, the selectors you provided in your example don't exist. They look autogenerated, so you'll need to use better ones.

<img width="400" alt="Screenshot 2024-02-17 at 11 17 08 PM" src="https://github.com/seleniumbase/SeleniumBase/assets/6788579/1cbae9fd-4edf-41f8-88fa-8dc5b05b61cb">

It looks like you're trying to do tab/window switching. See the example: [examples/test_window_switching.py](https://github.com/seleniumbase/SeleniumBase/blob/master/examples/test_window_switching.py)

Note that special pop-ups often appear in iframes. If so, you'll need to use special methods to switch into the iframes first. See the example: [examples/iframe_tests.py](https://github.com/seleniumbase/SeleniumBase/blob/master/examples/iframe_tests.py)

**Reply by ChowKwanhwa** [reply]
![image](https://github.com/seleniumbase/SeleniumBase/assets/30934104/89ccf036-f31b-41dd-814f-7daf3ddc4df2)

I am sorry. The Twitter authentication popup looks very different from the normal iframe popups. Because when I search for the css seletors from the popup in the main html searchbar, I can't find any results. It looks like some javascript injections. 

But if you want to see the popup window yourself on the page, you will have to scroll down and click on the buttons, it will show up after some dialogues. 
![image](https://github.com/seleniumbase/SeleniumBase/assets/30934104/a77ed9ac-6348-4401-9d92-48f1295bb165)

**Reply by mdmintz** [reply]
A direct link would be preferable for figuring out the script. Something like this: https://api.twitter.com/oauth/authenticate but without the missing details, and maybe in English so that I can read it more easily. This question is probably better suited for https://stackoverflow.com/
